src/components/JobsContext.jsx

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState
+} from 'react'
+import { JOBS_DOCTYPE } from '../doctypes'
+import CozyRealtime from 'cozy-realtime'
+import Alerter from 'cozy-ui/transpiled/react/Alerter'
+import logger from 'cozy-logger'
+import { useI18n } from 'cozy-ui/react/I18n'
+
+const log = logger.namespace('import.context')
+
+export const JobsContext = createContext({})
+
+export const useJobsContext = () => {
+  return useContext(JobsContext)
+}
+
+const JobsProvider = ({ children, client }) => {
+  const [jobsInProgress, setJobsInProgress] = useState([])
+  const { t } = useI18n()
+  const realtimeStartedRef = useRef(false)
+  const jobsInProgressRef = useRef([])
+
+  const realtime = useMemo(() => {
+    return new CozyRealtime({ client })
+  }, [client])
+
+  const handleRealtime = data => {
+    const { worker, state, message: msg } = data
+    const { current: currJobsInProgress } = jobsInProgressRef
```

Comment:
Pourquoi on a besoin de jobsInProgressRef et en même temps de jobsInProgress ? C'est pas clair pour moi, y-a-t il besoin d'un commentaire ou alors pourrait-on juste utiliser jobsInProgress ?

Reply:
Ah yes, this is because of the handleRealtime closure that is not updated when jobsInProgress changes.

Reply:
Yes exactly